source/common/http/filter_manager.cc

```diff
@@ -242,6 +242,10 @@ Router::RouteConstSharedPtr ActiveStreamFilterBase::route(const Router::RouteCal
   return parent_.filter_manager_callbacks_.route(cb);
 }
 
+void ActiveStreamFilterBase::setRoute(Router::RouteConstSharedPtr route) {
+  parent_.filter_manager_callbacks_.setRoute(route);
```

Comment:
`std::move(route)` to avoid a copy

Reply:
yup done, thanks for pointing these out! 

https://github.com/envoyproxy/envoy/pull/15266/commits/54afe69c91609d8365f22c73b74b1e7c40e4ba22